doc/howtos/web.rst

```diff
@@ -2252,45 +2363,44 @@ the most useful being:
 
         .. code-block:: javascript
 
-            local.WidgetCoordinates = instance.web.form.FormWidget.extend({
+            var WidgetCoordinates = Form.FormWidget.extend({
                 events: {
-                    'click button': function () {
+                    'click button': function() {
                         navigator.geolocation.getCurrentPosition(
                             this.proxy('received_position'));
                     }
                 },
                 start: function() {
-                    var sup = this._super();
-                    this.field_manager.on("field_changed:provider_latitude", this, this.display_map);
-                    this.field_manager.on("field_changed:provider_longitude", this, this.display_map);
-                    this.on("change:effective_readonly", this, this.display_map);
+                    this._super();
+                    this.field_manager.on("field_changed:latitude", this, this.display_map);
+                    this.field_manager.on("field_changed:longitude", this, this.display_map);
                     this.display_map();
-                    return sup;
                 },
                 display_map: function() {
                     this.$el.html(QWeb.render("WidgetCoordinates", {
-                        "latitude": this.field_manager.get_field_value("provider_latitude") || 0,
-                        "longitude": this.field_manager.get_field_value("provider_longitude") || 0,
+                        "latitude": this.field_manager.get_field_value("latitude") || 0,
+                        "longitude": this.field_manager.get_field_value("longitude") || 0,
                     }));
-                    this.$("button").toggle(! this.get("effective_readonly"));
                 },
                 received_position: function(obj) {
                     this.field_manager.set_values({
-                        "provider_latitude": obj.coords.latitude,
-                        "provider_longitude": obj.coords.longitude,
+                        "latitude": obj.coords.latitude,
+                        "longitude": obj.coords.longitude,
                     });
                 },
             });
 
-            instance.web.form.custom_widgets.add('coordinates', 'instance.oepetstore.WidgetCoordinates');
+            core.form_custom_registry.add('coordinates', WidgetCoordinates);
 
         .. code-block:: xml
 
             <t t-name="WidgetCoordinates">
                 <iframe width="400" height="300"
-                    t-attf-src="https://maps.google.com/?ie=UTF8&amp;ll={{latitude}},{{longitude}}&amp;output=embed">
+                    t-attf-src="https://www.google.com/maps/embed/v1/place?q={{latitude}},{{longitude}}&amp;key=AIzaSyC6vLt-bzYH0U6h1aZJmQee8jpxtcjeRVY">
```

Comment:
not sure you should include the key here (i guess maps doesn't work without an api key anymore though)